2018-02-14  Mingxi Sun  <classmsu@lnxsrv09.seas.ucla.edu>

	*lib/timespec.h changed subtraction to comparison to avoid integer overflow so super old files will be handled properly 

	

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2018-02-14 00:33:37.488236000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_nsec >b.tv_nsec)
+    return 1;
+  else if (a.tv_nsec ==b.tv_nsec)
+    return 0;
+  else
+    return -1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
